src/cpp/client/client_context.cc

```diff
@@ -41,9 +41,10 @@ class DefaultGlobalClientCallbacks final
 };
 
 static internal::GrpcLibraryInitializer g_gli_initializer;
-static DefaultGlobalClientCallbacks g_default_client_callbacks;
+static DefaultGlobalClientCallbacks* g_default_client_callbacks =
+    new DefaultGlobalClientCallbacks();
```

Comment:
I'm not sure that I'm happy with this PR as it looks like a leak to me. That said, I know the previous version was also noncompliant with our style guide since DefaultGlobalClientCallbacks isn't trivially destructible.

Reply:
It's not a truly leak here, g_default_client_callbacks is only initialize once, there's only one allocation of g_default_client_callbacks. Both previous solution & this one are cleaned when process exit. This solution is cleaned when process is cleaned by kernel not as global variables. For this case, this solution is not overhead and clean enough for me. Probably another solution is g_default_client_callbacks is member variable of ClientContext, then could be destroyed as ClientContext destructed. Such as follows:

```
ClientContext::ClientContext()
    : initial_metadata_received_(false),
      wait_for_ready_(false),
      wait_for_ready_explicitly_set_(false),
      idempotent_(false),
      cacheable_(false),
      call_(nullptr),
      call_canceled_(false),
      deadline_(gpr_inf_future(GPR_CLOCK_REALTIME)),
      census_context_(nullptr),
      propagate_from_call_(nullptr),
      initial_metadata_corked_(false) {
  if (g_client_callbacks == nullptr) {
    g_client_callbacks = &default_client_callbacks;
  }
  g_client_callbacks->DefaultConstructor(this);
}
```
default_client_callbacks is member variable in ClientContext.
This solution would introduce 8 more bytes in ClientContext and one branch prediction is constructor in ClientContext.

Reply:
@vjpai what's your prefer?